lang/block.go

```diff
@@ -109,24 +118,53 @@ func ParseBlocks(cfg *Config, blocks []comment.Block, inline bool) []*Block {
 }
 
 func printText(b *strings.Builder, text ...comment.Text) {
-	for i, t := range text {
-		if i > 0 {
-			b.WriteRune(' ')
-		}
-
+	for _, t := range text {
 		switch v := t.(type) {
 		case comment.Plain:
 			b.WriteString(string(v))
 		case comment.Italic:
 			b.WriteString(string(v))
 		case *comment.DocLink:
-			printText(b, v.Text...)
+			b.WriteString(printDocLink(v))
 		case *comment.Link:
-			printText(b, v.Text...)
+			b.WriteString(fmt.Sprintf("%s(%s)", v.Text, v.URL))
 		}
 	}
 }
 
+func printDocLink(docLink *comment.DocLink) string {
+	// Case [Volume]
+	text := fmt.Sprintf("%s", docLink.Text)
+	if docLink.ImportPath == "" {
+		return printLocalLink(text, fmt.Sprintf("Type %s", docLink.Name))
+	}
+
+	// Case [core.Volume]
+	if docLink.ImportPath == actualPackage {
+		return printLocalLink(text, fmt.Sprintf("Type %s", docLink.Name))
+	}
+
+	// Case [os.File] or [repourl/orga/package.type]
+	if docLink.Name != "" {
+		return fmt.Sprintf("%s(%s/%s#%s)", text, urlPckgGoDev, docLink.ImportPath, docLink.Name)
+	}
+	return fmt.Sprintf("%s(%s/%s)", text, urlPckgGoDev, docLink.ImportPath)
+}
+
+// TODO Local links are fixed in github format.
+// For other formats break extend block parsing.
+// Steps:
+// Separate paragraphs blocks with filtered [comment.DocLink]. -> new Block types DocLink
+// Extend template with this type.
```

Comment:
I fail to grasp the comment. Do you try to hint what the next developer must do to solve the TODO? If so I suggest to create another issue and describe the action there (I just added the github issues capability for this project)

If not we should rethink this comment.

Reply:
https://github.com/cloudogu/gomarkdoc/issues/3